src/inc/til/rect.h

```diff
@@ -744,6 +815,20 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
                 gsl::narrow<short>(bottom - 1),
             };
         }
+
+        // NOTE: This will convert from INCLUSIVE on the way in because
+        // that is generally how SMALL_RECTs are handled in console code and via the APIs.
+        explicit constexpr rect(const inclusive_rect other) noexcept :
+            rect{ other.Left, other.Top, other.Right + 1, other.Bottom + 1 }
+        {
+        }
+
+        // NOTE: This will convert back to INCLUSIVE on the way out because
+        // that is generally how SMALL_RECTs are handled in console code and via the APIs.
+        constexpr inclusive_rect to_inclusive_rect() const
+        {
+            return { left, top, right - 1, bottom - 1 };

```

Comment:
The conversion between `rect` and `inclusive_rect` performs no overflow checks.
I personally consider that an acceptable risk, since we simultaneously upgrade from 16 to 32 bit integers for all intermediate coordinate representations like this one.
What about you all?

Reply:
Yes that sounds fine, but you should perhaps document that assumption in the comment.